<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainWindowGui</class>
 <widget class="QMainWindow" name="mainWindowGui">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>953</width>
    <height>601</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="mainWidget">
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="0">
       <widget class="OpenGLCanvas" name="graphics">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSlider" name="verticalSlider">
        <property name="maximumSize">
         <size>
          <width>18</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QSlider" name="horizontalSlider">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>18</height>
         </size>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="label_14">
          <property name="minimumSize">
           <size>
            <width>60</width>
            <height>28</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>60</width>
            <height>28</height>
           </size>
          </property>
          <property name="text">
           <string>Vertices</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_15">
          <property name="minimumSize">
           <size>
            <width>47</width>
            <height>28</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>47</width>
            <height>28</height>
           </size>
          </property>
          <property name="text">
           <string>Object</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QComboBox" name="comboBox">
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>25</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>25</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QListWidget" name="listWidget">
          <property name="minimumSize">
           <size>
            <width>268</width>
            <height>100</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>268</width>
            <height>100</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QTabWidget" name="tabWidget">
          <property name="minimumSize">
           <size>
            <width>280</width>
            <height>278</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>280</width>
            <height>278</height>
           </size>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab">
           <attribute name="title">
            <string>PhongModel</string>
           </attribute>
           <widget class="QDoubleSpinBox" name="doubleSpinBox">
            <property name="geometry">
             <rect>
              <x>40</x>
              <y>20</y>
              <width>62</width>
              <height>27</height>
             </rect>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.050000000000000</double>
            </property>
            <property name="value">
             <double>0.500000000000000</double>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
            <property name="geometry">
             <rect>
              <x>40</x>
              <y>50</y>
              <width>62</width>
              <height>27</height>
             </rect>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.050000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="doubleSpinBox_3">
            <property name="geometry">
             <rect>
              <x>40</x>
              <y>80</y>
              <width>62</width>
              <height>27</height>
             </rect>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.050000000000000</double>
            </property>
            <property name="value">
             <double>0.500000000000000</double>
            </property>
           </widget>
           <widget class="QLabel" name="label">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>30</y>
              <width>21</width>
              <height>17</height>
             </rect>
            </property>
            <property name="text">
             <string>Ka</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_2">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>60</y>
              <width>21</width>
              <height>17</height>
             </rect>
            </property>
            <property name="text">
             <string>Kd</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_3">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>90</y>
              <width>21</width>
              <height>17</height>
             </rect>
            </property>
            <property name="text">
             <string>Ks</string>
            </property>
           </widget>
           <widget class="QSpinBox" name="spinBox">
            <property name="geometry">
             <rect>
              <x>210</x>
              <y>20</y>
              <width>55</width>
              <height>27</height>
             </rect>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
           </widget>
           <widget class="QLabel" name="label_4">
            <property name="geometry">
             <rect>
              <x>120</x>
              <y>30</y>
              <width>91</width>
              <height>17</height>
             </rect>
            </property>
            <property name="text">
             <string>Phong Const</string>
            </property>
           </widget>
           <widget class="QCheckBox" name="checkBox">
            <property name="geometry">
             <rect>
              <x>160</x>
              <y>60</y>
              <width>102</width>
              <height>22</height>
             </rect>
            </property>
            <property name="text">
             <string>Half Toning</string>
            </property>
           </widget>
          </widget>
          <widget class="QWidget" name="tab_2">
           <attribute name="title">
            <string>Light</string>
           </attribute>
           <widget class="QDoubleSpinBox" name="doubleSpinBox_4">
            <property name="geometry">
             <rect>
              <x>20</x>
              <y>20</y>
              <width>62</width>
              <height>27</height>
             </rect>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="value">
             <double>100.000000000000000</double>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="doubleSpinBox_5">
            <property name="geometry">
             <rect>
              <x>20</x>
              <y>50</y>
              <width>62</width>
              <height>27</height>
             </rect>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="doubleSpinBox_6">
            <property name="geometry">
             <rect>
              <x>20</x>
              <y>80</y>
              <width>62</width>
              <height>27</height>
             </rect>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
           </widget>
           <widget class="QLabel" name="label_5">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>30</y>
              <width>16</width>
              <height>17</height>
             </rect>
            </property>
            <property name="text">
             <string>X</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_6">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>60</y>
              <width>16</width>
              <height>17</height>
             </rect>
            </property>
            <property name="text">
             <string>Y</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_7">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>90</y>
              <width>16</width>
              <height>17</height>
             </rect>
            </property>
            <property name="text">
             <string>Z</string>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="doubleSpinBox_7">
            <property name="geometry">
             <rect>
              <x>210</x>
              <y>20</y>
              <width>62</width>
              <height>27</height>
             </rect>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>0.500000000000000</double>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="doubleSpinBox_8">
            <property name="geometry">
             <rect>
              <x>210</x>
              <y>50</y>
              <width>62</width>
              <height>27</height>
             </rect>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>0.500000000000000</double>
            </property>
           </widget>
           <widget class="QLabel" name="label_8">
            <property name="geometry">
             <rect>
              <x>110</x>
              <y>60</y>
              <width>101</width>
              <height>17</height>
             </rect>
            </property>
            <property name="text">
             <string>Light Intensity</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_9">
            <property name="geometry">
             <rect>
              <x>90</x>
              <y>30</y>
              <width>121</width>
              <height>17</height>
             </rect>
            </property>
            <property name="text">
             <string>Ambient Intensity</string>
            </property>
           </widget>
          </widget>
         </widget>
        </item>
        <item row="3" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>178</width>
            <height>15</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>953</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionClose"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionTranslate">
   <property name="text">
    <string>Translate</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OpenGLCanvas</class>
   <extends>QGraphicsView</extends>
   <header>OpenGLCanvas.h</header>
   <slots>
    <signal>MousePositionChanged(QPoint)</signal>
    <signal>SizeChanged(QSize)</signal>
    <slot>SetPixelPositionX(int)</slot>
    <slot>SetPixelPositionY(int)</slot>
    <slot>drawEverything(Polygon*)</slot>
    <slot>refreshDraw()</slot>
    <slot>setClipMinX(int)</slot>
    <slot>setClipMinY(int)</slot>
    <slot>setClipMaxX(int)</slot>
    <slot>setClipMaxY(int)</slot>
    <slot>setDDA(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>mainWindowGui</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>194</x>
     <y>144</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>mainWindowGui</receiver>
   <slot>_FileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>194</x>
     <y>144</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>graphics</receiver>
   <slot>SetPixelPositionX(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>89</x>
     <y>476</y>
    </hint>
    <hint type="destinationlabel">
     <x>67</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>verticalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>graphics</receiver>
   <slot>SetPixelPositionY(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>206</y>
    </hint>
    <hint type="destinationlabel">
     <x>258</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>graphics</sender>
   <signal>MousePositionChanged(QPoint)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>CanvasMousePositionChanged(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>271</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>graphics</sender>
   <signal>SizeChanged(QSize)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>CanvasSizeChanged(QSize)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>155</x>
     <y>138</y>
    </hint>
    <hint type="destinationlabel">
     <x>271</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mainWindowGui</sender>
   <signal>refresh()</signal>
   <receiver>graphics</receiver>
   <slot>refreshDraw()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>271</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>155</x>
     <y>138</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>selectObj(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>452</x>
     <y>49</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_4</sender>
   <signal>valueChanged(double)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>updateLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>471</x>
     <y>240</y>
    </hint>
    <hint type="destinationlabel">
     <x>755</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_5</sender>
   <signal>valueChanged(double)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>updateLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>496</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>758</x>
     <y>219</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_6</sender>
   <signal>valueChanged(double)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>updateLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>480</x>
     <y>302</y>
    </hint>
    <hint type="destinationlabel">
     <x>758</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_7</sender>
   <signal>valueChanged(double)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>updateLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>656</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel">
     <x>721</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_8</sender>
   <signal>valueChanged(double)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>updateLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>671</x>
     <y>264</y>
    </hint>
    <hint type="destinationlabel">
     <x>720</x>
     <y>284</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>updatePhong()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>498</x>
     <y>241</y>
    </hint>
    <hint type="destinationlabel">
     <x>742</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_2</sender>
   <signal>valueChanged(double)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>updatePhong()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>511</x>
     <y>266</y>
    </hint>
    <hint type="destinationlabel">
     <x>733</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_3</sender>
   <signal>valueChanged(double)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>updatePhong()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>500</x>
     <y>291</y>
    </hint>
    <hint type="destinationlabel">
     <x>735</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>updatePhong()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>661</x>
     <y>242</y>
    </hint>
    <hint type="destinationlabel">
     <x>740</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>mainWindowGui</receiver>
   <slot>updateHalfTone(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>627</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>742</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>openTheFile(QString)</signal>
  <signal>updateGraphics(Polygon*)</signal>
  <signal>refresh()</signal>
  <slot>_Exit()</slot>
  <slot>_FileOpen()</slot>
  <slot>CanvasMousePositionChanged(QPoint)</slot>
  <slot>CanvasSizeChanged(QSize)</slot>
  <slot>selectObj(int)</slot>
  <slot>setTranslX(double)</slot>
  <slot>setTranslY(double)</slot>
  <slot>setScaleX(double)</slot>
  <slot>setTheta(double)</slot>
  <slot>setScaleY(double)</slot>
  <slot>translateObj()</slot>
  <slot>scaleObj()</slot>
  <slot>rotateObj()</slot>
  <slot>setClipMaxXMin(int)</slot>
  <slot>setClipMaxYMin(int)</slot>
  <slot>clipChange()</slot>
  <slot>setAxis1()</slot>
  <slot>setAxis2()</slot>
  <slot>setTranslZ(double)</slot>
  <slot>setScaleZ(double)</slot>
  <slot>updatePhong()</slot>
  <slot>updateLight()</slot>
  <slot>updateHalfTone(bool)</slot>
 </slots>
</ui>
